includes/framework/QDbBackedSessionHandler.class.php

```diff
@@ -123,6 +123,15 @@ public static function SessionRead($id) {
 			if (!$result_row) // either the data was empty or the row was not found
 				return '';
 			$strData = $result_row['data'];
+
+			if(strstr($objDatabase->Adapter, 'PostgreSql')) {
```

Comment:
@spekary I understand your point here. Lets look on it from other side:
1. here we obtain data some database
2. we know that the data is a `blob`

It looks like we need some database-agnostic method to obtain typed data. E.g. 

```php
$objDatabase->getValue($strData, QDatabaseFieldType::Blob);
```

It looks like data-conversion layer to be refactored from the DB layer and exposed to the end user maybe.

It is important issue for me, because any `if`-s like that in the codebase are a bad smell of architectural flaws. In this case it is simply a sign that some functionality is over-incapsulated here.